### PR TITLE
[kubeapiserver] Allow unbundling of events

### PR DESCRIPTION
**Pull request by juliogreff**
### What does this PR do?

When `unbundled_events` is set to `true` in the `kubernetes_apiserver`
integration, Kubernetes events are no longer bundled together by
InvolvedObject, and instead each will generate a separate Datadog event.

### Describe how to test/QA your changes

With `unbundle_events` in `conf.d/kubernetes_apiserver.d/conf.yaml` set to `true` and `false`, check that kubernetes event collection works as expected in the DCA.

Also test that the different types of filters work as expected. Filters were changed in #13484, so check that PR for how to use them.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/agent/dist/conf.d/kubernetes_apiserver.d/conf.yaml.example [hunk 1]
```diff
@@ -12,8 +12,36 @@ instances:
     #   - <KEY_1>:<VALUE_1>
     #   - <KEY_2>:<VALUE_2>
 
+    ## @param unbundle_events - boolean - optional - default: false
+    ## Makes each collected Kubernetes event be transformed into its own
```
Comment (maycmlee) [inline cmd/agent/dist/conf.d/kubernetes_apiserver.d/conf.yaml.example:16]:
```suggestion
    ## Transforms each collected Kubernetes event into its own
```

File: cmd/agent/dist/conf.d/kubernetes_apiserver.d/conf.yaml.example [hunk 2]
```diff
@@ -12,8 +12,36 @@ instances:
...
+    ## Datadog event, instead of grouping them together by InvolvedObject.
+    #
+    # unbundle_events: true
+
+    ## @param collected_event_types - map of array of strings - optional
+    ## Specify which events to be collected. Map of Kind as key, array of event
+    ## reasons as value. Keys are case-insensitive, values are case-sensitive.
+    ## Only effective is unbundle_events is true, otherwise filtered_event_types
```
Comment (maycmlee) [inline cmd/agent/dist/conf.d/kubernetes_apiserver.d/conf.yaml.example:24]: Does "Only effective is unbundle_events is true" mean it's only effective when unbundle_events is true? If so, I'd suggest rewording to:
"Only effective when unbundle_events is true..."

File: cmd/agent/dist/conf.d/kubernetes_apiserver.d/conf.yaml.example [hunk 3]
```diff
@@ -12,8 +12,36 @@ instances:
...
+        - FailedAttachVolume
+      node:
+        - TerminatingEvictedPod
+        - NodeNotReady
+        - Rebooted
+        - HostPortConflict
+
     ## @param filtered_event_types - array of strings - optional
     ## Specify a list of exclusion filters over the event type, involvedObject.kind, reason, following the Kubernetes field-selector format.
+    ## Only effective is unbundle_events is false, otherwise collected_event_types
```
Comment (maycmlee) [inline cmd/agent/dist/conf.d/kubernetes_apiserver.d/conf.yaml.example:43]: Same as above

**Review by maycmlee — APPROVED** [review]
Approving for docs, but left a couple of suggestions.

**Review by djmitche — APPROVED** [review]
`cmd/agent/dist/conf.d/kubernetes_apiserver.d/conf.yaml.example` lgtm

File: pkg/collector/corechecks/cluster/kubernetesapiserver/events_common.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+		return metrics.EventAlertTypeWarning
+	default:
+		log.Debugf("Unknown event type '%s'", k8sType)
+		return metrics.EventAlertTypeInfo
+	}
+}
+
+func getInvolvedObjectTags(involvedObject v1.ObjectReference) []string {
+	tags := []string{
+		fmt.Sprintf("kubernetes_kind:%s", involvedObject.Kind),
```
Comment (clamoriniere) [inline pkg/collector/corechecks/cluster/kubernetesapiserver/events_common.go:50]: usually we use `kube_` prefix and not `kubernetes_` 
If `kubernetes_kind` is already use, you can keep it, but `kube_kind` can be added.

File: pkg/collector/corechecks/cluster/kubernetesapiserver/events_common.go [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+		fmt.Sprintf("name:%s", involvedObject.Name),
```
Comment (clamoriniere) [inline pkg/collector/corechecks/cluster/kubernetesapiserver/events_common.go:51]: maybe here the same. `kube_name` can be added

**Review by clamoriniere — COMMENTED** [review]
Sorry for the delay in the Review. 
Some minor comments. 
I like the refactoring and how it is easier to follow the code. 💯

**Review by clamoriniere — APPROVED** [review]
Awesome 💯
